src/throttler/throttler.ts

```diff
@@ -254,4 +230,79 @@ export abstract class Throttler<T, R> {
       return p.resolve();
     });
   }
+
+  private initializeTimeout(taskData: TaskData<T, R>, cursorIndex: number): Promise<void> {
+    let rejectFunc: (err: TaskError) => void;
+    const timeoutMillis = taskData.timeoutMillis!;
+    const timeoutPromise = new Promise<void>((_, reject) => {
+      rejectFunc = reject;
+    });
+    taskData.timeoutId = setTimeout(() => {
+      taskData.isTimedOut = true;
+      rejectFunc(new TimeoutError(this.taskName(cursorIndex), timeoutMillis));
+    }, timeoutMillis);
+
+    return timeoutPromise;
+  }
+
+  private async executeTask(taskData: TaskData<T, R>, cursorIndex: number): Promise<any> {
+    const t0 = Date.now();
+    try {
+      const result = await this.handler(taskData.task);
+      const dt = Date.now() - t0;
+      if (dt < this.min) {
+        this.min = dt;
+      }
+      if (dt > this.max) {
+        this.max = dt;
+      }
+      this.avg = (this.avg * this.complete + dt) / (this.complete + 1);
+
+      this.success++;
+      this.complete++;
+      this.active--;
+      return result;
+    } catch (error) {
```

Comment:
nit: Convention is to use the name `err`.

Reply:
Done